### PR TITLE
Change ordering of blocks as required by SPIRV Spec

### PR DESCRIPTION
**Pull request by kkyzylova**
According to specification:

> The order of blocks in a function must satisfy the rule that blocks appear before all blocks they dominate.

This pass helps to avoid the cases which are contrary to SPIRV Spec.

### CODE REVIEW

**Comment by yxsamliu** [comment]
Do we have a test for this?

Also, is this sufficient to always satisfy the spec? I am wondering if we need a dedicated pass to do this.

**Comment by kkyzylova** [comment]
I've changed the fix and added a regression test, please take a look. Thanks.
There is no SimplifyCFG pass now.

File: lib/SPIRV/SPIRVRegularizeLLVM.cpp [hunk 1]
```diff
@@ -154,6 +156,34 @@ SPIRVRegularizeLLVM::regularize() {
         }
       }
     }
+
```
Comment (yxsamliu) [inline lib/SPIRV/SPIRVRegularizeLLVM.cpp:161]: Please refactor this part as a member function and add come comments about how it works.

I guess this only fixes the specific issue for switch instruction, not a generic transformation to guarantee that blocks appearing before all blocks they dominate, right? If so it is OK, just mention that in the comment and give it a proper name e.g. fixBlockOrderForSwitch.

**Comment by yxsamliu** [comment]
Please update the title for this pull request for what it does. Thanks.

**Comment by yxsamliu** [comment]
It caused travis-ci failure. Can you fix that? Thanks.

**Pull request merged**